walk/config.go

```diff
@@ -119,22 +133,38 @@ func (*Configurer) KnownDirectives() []string {
 }
 
 func (cr *Configurer) Configure(c *config.Config, rel string, f *rule.File) {
-	wc := getWalkConfig(c)
-	wcCopy := &walkConfig{}
-	*wcCopy = *wc
-	wcCopy.ignore = false
+	if c.Exts[walkNameCached] != nil {
+		// A normal Configurer implementation would process directives and set
+		// c.Exts[walkName] here. However, we've parallelized the tree walk and
+		// processed the configuration ahead of time in configureForWalk. So instead,
+		// the caller of this method (configure) sets c.Exts[walkNameCache] to the
+		// preprocessed configuration. We copy it to c.Exts[walkName] instead of
+		// re-processing directives.
+		c.Exts[walkName] = c.Exts[walkNameCached]
```

Comment:
Could we delete the `walkNameCached` key here to save memory?

Reply:
Done, but note this only removes the map entry, an alias to the configuration. It won't save memory unless `c.Exts` is rehashed.